gnucash/import-export/test/gtest-import-backend.cpp

```diff
@@ -0,0 +1,259 @@
+#include <gtk/gtk.h>
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+
+#include "config.h"
+
+#include "AccountP.h"
+#include "guid.hpp"
+
+extern "C"
+{
+#include "cashobjects.h"
+#include "gnc-locale-utils.h"
+#include "gnc-prefs-p.h"
+#include "../import-backend.h"
+}
+
+
+// mock up for PrefsBackend (singleton class)
+class MockPrefsBackend : PrefsBackend
+{
+public:
+    MockPrefsBackend(MockPrefsBackend const&) = delete;
+    MockPrefsBackend& operator=(MockPrefsBackend const&) = delete;
+
+    static MockPrefsBackend* getInstance()
+    {
+        if (prefsbackend == NULL)
+        {
+            prefsbackend = new MockPrefsBackend();
+            prefsbackend->get_bool = _get_bool;
+        }
+        else if (prefsbackend->get_bool != _get_bool)
+        {
+            // prefsbackend is already assigned to another implementation
+            return nullptr;
+        }
+        testing::Mock::AllowLeak((MockPrefsBackend*)prefsbackend);
+        return (MockPrefsBackend*)prefsbackend;
+    }
+
+    MOCK_METHOD2(getBool, bool(const gchar* group, const gchar* pref_name));
+
+private:
+    MockPrefsBackend() {}
+    ~MockPrefsBackend() {}
+
+    static gboolean _get_bool(const gchar* group, const gchar* pref_name)
+    {
+        return (gboolean)((MockPrefsBackend*)prefsbackend)->getBool(group, pref_name);
+    }
+};
+
+
+// mock up for GncImportMatchMap
+class MockGncImportMatchMap : public GncImportMatchMap
+{
+public:
+    MOCK_METHOD2(findAccount, Account*(const char*, const char*));
+    MOCK_METHOD1(findAccountBayes, Account*(std::vector<std::string>));
+};
+
+
+/* overwrite functions for GncImportMatchMap */
+
+Account* gnc_account_imap_find_account (
+        GncImportMatchMap *imap,
+        const char* category,
+        const char *key)
+{
+    return ((MockGncImportMatchMap*)imap)->findAccount(category, key);
+}
+
+Account* gnc_account_imap_find_account_bayes (
+        GncImportMatchMap *imap,
+        GList *tokens)
+{
+    std::vector<std::string> tokenVec;
+
+    for (auto token = tokens; token; token = token->next)
+    {
+        // remove duplicated tokens by transforming list of tokens into a set
```

Comment:
`std::vector` != `std::set`. `std::vector` doesn't remove duplicates.

Reply:
Oh, this comment was a leftover from my first try and is not correct anymore. I removed it.